### DISCUSSION TITLE
Unsubscribe subscription [Rest API] working?

### DISCUSSION THREAD

**Discussion by trevordowdle** [apache/pulsar#17379] [Q&A] [answered]
I tried unsubscribing a subscription on a topic via the rest api as indicated here:
https://pulsar.apache.org/docs/admin-api-topics/#unsubscribe-subscription

But it didn't work and the response I got was along the lines of "not found".

I then tried the following to unsubscribe a subscription on a whole namespace and this worked for me, but the above didn't.
https://pulsar.apache.org/admin-rest-api/#operation/unsubscribeNamespace

I was just wondering if anyone could confirm for me that [this](https://pulsar.apache.org/docs/admin-api-topics/#unsubscribe-subscription) works to unsubscribe a subscription on a single topic.

I'm running ~ Pulsar version 2.7.2

Thanks

**Comment by michaeljmarshall** [comment]
Are you able to provide the exact error you observed? Were you using the rest api or the Pulsar Admin CLI? Also, can you confirm that the subscription exists on the topic by calling `pulsar-admin topics stats <topic-name>`?

**Reply by trevordowdle** [reply]
Sorry here is the link to my reply.

https://github.com/apache/pulsar/discussions/17379#discussioncomment-3529738

**Comment by trevordowdle** [comment]
Thank you for your response.

I am using the rest api, here is a sample of the end point:

"DELETE" request to:
http://pulsar-test.com:8080/admin/v2/namespaces/test/ingest/test/subscription/subscription-message-save

Result:
statusText = 'Not Found'
status = 404

Running: pulsar-admin topics stats persistent://test/ingest/test show the subscription still active.
```
  "subscriptions" : {
    "subscription-message-save" : {
      "msgRateOut" : 0.0,
      "msgThroughputOut" : 0.0,
      "bytesOutCounter" : 0,
      "msgOutCounter" : 0,
      "msgRateRedeliver" : 0.0,
      "chuckedMessageRate" : 0,
      "msgBacklog" : 0,
      "backlogSize" : 0,
      "msgBacklogNoDelayed" : 0,
      "blockedSubscriptionOnUnackedMsgs" : false,
      "msgDelayed" : 0,
      "unackedMessages" : 0,
      "type" : "Exclusive",
      "msgRateExpired" : 0.0,
      "totalMsgExpired" : 0,
      "lastExpireTimestamp" : 0,
      "lastConsumedFlowTimestamp" : 1662056861336,
      "lastConsumedTimestamp" : 0,
      "lastAckedTimestamp" : 0,
      "lastMarkDeleteAdvancedTimestamp" : 0,
      "consumers" : [ ],
      "isDurable" : true,
      "isReplicated" : false,
      "consumersAfterMarkDeletePosition" : { },
      "nonContiguousDeletedMessagesRanges" : 0,
      "nonContiguousDeletedMessagesRangesSerializedSize" : 0
    }
  },
```

What works:

"Post" request to
http://pulsar-test.com:8080/admin/v2/namespaces/test/ingest/unsubscribe/subscription-message-save

Result:
statusText = 'No Content'
status = 204

Running: pulsar-admin topics stats persistent://test/ingest/test shows the subscription has been successfully removed.
`"subscriptions" : { },`

**Reply by michaeljmarshall** [reply]
Thanks, that helps a ton.

**Comment by michaeljmarshall** [comment, marked as answer]
It looks like that documentation is incorrect. I also get a 404 when I call that endpoint. However, when I run `bin/pulsar-admin topics unsubscribe -s sub test`, I noticed that the broker logs a different endpoint:

> 23:49:23.569 [BookKeeperClientWorker-OrderedExecutor-0-0] INFO  org.eclipse.jetty.server.RequestLog - 127.0.0.1 - - [01/Sep/2022:23:49:23 -0500] "PUT /admin/v2/persistent/public/default/test/subscription/sub HTTP/1.1" 204 0 "-" "Pulsar-Java-v2.7.2" 78

The correct endpoint includes `/persistent`. If you try `/admin/v2/persistent/{tenant}/{namespace}/{topic}/subscription/{subName}`, it should work.

Would you mind creating an issue in the project to get the documentation fixed? Thanks!

**Reply by trevordowdle** [reply]
Thank you, yes I will try this and get back to you and then I'll create an issue in the project to get the documentation fixed.

Had a busy week and haven't got around to it yet, but will soon.
